src/go/service/sctpd/proxy_uplink_server.go

```diff
@@ -0,0 +1,63 @@
+// Copyright 2021 The Magma Authors.
+//
+// This source code is licensed under the BSD-style license found in the
+// LICENSE file in the root directory of this source tree.
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sctpd
+
+import (
+	"context"
+	"net"
+
+	"google.golang.org/grpc"
+
+	"github.com/magma/magma/log"
+	"github.com/magma/magma/protos/magma/sctpd"
+)
+
+type ProxyUplinkServer struct {
+	client sctpd.SctpdUplinkClient
+
+	log.Logger
+	*sctpd.UnimplementedSctpdUplinkServer
+}
+
+func NewProxyUplinkServer(logger log.Logger, cc *grpc.ClientConn) *ProxyUplinkServer {
+	return &ProxyUplinkServer{Logger: logger, client: sctpd.NewSctpdUplinkClient(cc)}
+}
+
+func (p *ProxyUplinkServer) SendUl(ctx context.Context, req *sctpd.SendUlReq) (*sctpd.SendUlRes, error) {
```

Comment:
My first question(s) are whether this is a blocking call or not - would be helpful for SendUl godoc to specify blocking or nonblocking nature.  Second question generally is the thread safety of the ProxyUplinkServer generally (can anybody bang on SendUl from whatever context they chose?). This topic might belong as godoc on ProxyUplinkServer?

Reply:
ah, that's kind of a grpc server impl detail, but by default all unary calls are concurrent/blocking.

the server spins a goroutine for each inbound request and executes the provided implementation.

this "simple" pattern is the recommended default for all our stuff going foward: do your work in a linear/blocking fashion and count on the underlying server to spin goroutines to allow for parallel processing (vs. MME's current strategy of attempting to serialize work in queues; this is somewhat like concurrent vs. co-operative threading patterns)